### PR TITLE
Adds log stacking to admin logs

### PR DESCRIPTION
**Pull request by FlattestGuitar**
When two logs that are made in less than ten seconds apart get created, they instead clump together into a single log with a number, like this:

Any log that isn't perfectly the same as its neighbour won't get merged and will split them, so a fight with both parties involved will still look like a chessboard.

No changelog, because this is all backstage.

Also, taking suggestions on how to make the proc readable. I've added enough comments, I think.

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
Kreygasm

File: code/modules/mob/mob_defines.dm [hunk 1]
```diff
@@ -74,6 +74,7 @@
 	var/overeatduration = 0		// How long this guy is overeating //Carbon
 	var/intent = null//Living
 	var/shakecamera = 0
+	var/a_intent = I_HELP//Living
```
Comment (FlattestGuitar) [inline code/modules/mob/mob_defines.dm]: 🤦‍♂️

**Comment by fludd12** [comment]
Personally, I'd prefer the multiplier number to be at the end so it doesn't blend in with the time but this is amazing nonetheless.

**Comment by Regen1** [comment]
Agree with fludd, i'd like the counter at the end, it makes the flow a lot better

"this person attacked this person X amount"

**Comment by Fox-McCloud** [comment]
if I'm being pendantic..which I am

Makes it xN, instead of Nx.

**Comment by FlattestGuitar** [comment]
All of these changes, while purely cosmetic, will make the whole proc there a lot more convoluted, considering the changing position of the x.

The point right now is that we *know* there will only be 12 digits and everything that comes after those but before the first X will be the multipliers.

I caaaaan change it, but it won't really do anything.

**Comment by Fox-McCloud** [comment]
Yeahhh, I can see that. Don't worry about it.

File: code/datums/spell.dm [hunk 2]
```diff
@@ -162,7 +162,7 @@ var/list/spells = typesof(/obj/effect/proc_holder/spell) //needed for the badmin
 /obj/effect/proc_holder/spell/proc/perform(list/targets, recharge = 1, mob/user = usr) //if recharge is started is important for the trigger spells
 	before_cast(targets)
 	invocation()
-	user.attack_log += text("\[[time_stamp()]\] <font color='red'>[user.real_name] ([user.ckey]) cast the spell [name].</font>")
```
Comment (Fox-McCloud) [inline code/datums/spell.dm]: this should probably be under a 

`If(user && user.ckey)`

File: code/modules/mob/mob_defines.dm [hunk 3]
```diff
@@ -73,14 +74,14 @@
...
 	var/shakecamera = 0
-	var/a_intent = I_HELP//Living
 	var/m_intent = "run"//Living
 	var/lastKnownIP = null
 	var/atom/movable/buckled = null//Living
 	var/obj/item/l_hand = null//Living
 	var/obj/item/r_hand = null//Living
 	var/obj/item/weapon/back = null//Human/Monkey
 	var/obj/item/weapon/tank/internal = null//Human/Monkey
+	var/a_intent = I_HELP//Living
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_defines.dm]: Don't move this. Keep the intents together.

Comment (FlattestGuitar) [inline code/modules/mob/mob_defines.dm]: I've no idea how it got there. Fixed.

**Pull request merged**